sw/airborne/modules/multi/follow.c

```diff
@@ -47,22 +45,32 @@
 #define FOLLOW_OFFSET_Z 0.0
 #endif
 
-void follow_init(void)
-{
+#ifndef FOLLOW_AC_ID
+#error "Please define FOLLOW_AC_ID"
+#endif
 
-}
+#ifndef FOLLOW_WAYPOINT_ID
+#error "Please define FOLLOW_WAYPOINT_ID"
+#endif
+
+void follow_init(void){}
 
-void follow_change_wp(unsigned char *buffer)
+/*
+ * follow_wp(void)
+ * updates the FOLLOW_WAYPOINT_ID to a fixed offset from the last received location
+ * of other aircraft with id FOLLOW_AC_ID
+ */
+void follow_wp(void)
 {
-  struct EcefCoor_i new_pos;
-  struct EnuCoor_i enu;
-  new_pos.x = DL_REMOTE_GPS_ecef_x(buffer);
-  new_pos.y = DL_REMOTE_GPS_ecef_y(buffer);
-  new_pos.z = DL_REMOTE_GPS_ecef_z(buffer);
-
-  // Translate to ENU
-  enu_of_ecef_point_i(&enu, &state.ned_origin_i, &new_pos);
-  INT32_VECT3_SCALE_2(enu, enu, INT32_POS_OF_CM_NUM, INT32_POS_OF_CM_DEN);
+  struct ac_info_ * ac = get_ac_info(FOLLOW_AC_ID);
+
+  struct UtmCoor_f my_pos = utm_float_from_gps(&gps, 0);
```

Comment:
Better not depend on GPS here... just use position from state interface if actually needed...